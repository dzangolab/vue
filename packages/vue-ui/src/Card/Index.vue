<template>
  <div class="card">
    <header v-if="$slots.header || title" class="card-header">
      <slot name="header">
        <h3 class="card-title">
          {{ title }}
        </h3>
      </slot>
    </header>

    <div class="card-body">
      <slot></slot>
    </div>

    <footer v-if="$slots.footer" class="card-footer">
      <slot name="footer"></slot>
    </footer>
  </div>
</template>

<script lang="ts">
export default {
  name: "Card",
};
</script>

<script setup lang="ts">
defineProps({
  title: {
    type: String,
    default: undefined,
  },
});
</script>
