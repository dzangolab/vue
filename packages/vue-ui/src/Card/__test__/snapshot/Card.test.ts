import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Card from "../../Index.vue";

describe("Card", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Card, {
      slots: {
        default: "<p>This is a card body.</p>",
        header: "<h3>Test</h3>",
        footer: "This is a footer.",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
