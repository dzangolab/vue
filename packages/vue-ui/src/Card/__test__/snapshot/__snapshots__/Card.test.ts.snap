// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > matches snapshot 1`] = `
<div
  class="card"
>
  <header
    class="card-header"
  >
    
    <h3>
      Test
    </h3>
    
  </header>
  <div
    class="card-body"
  >
    
    <p>
      This is a card body.
    </p>
    
  </div>
  <footer
    class="card-footer"
  >
    
    This is a footer.
    
  </footer>
</div>
`;
